src/main/java/org/wildfly/security/auth/server/SecurityDomain.java

```diff
@@ -81,6 +83,7 @@
         this.principalDecoder = builder.principalDecoder;
         this.securityIdentityTransformer = builder.securityIdentityTransformer;
         this.trustedSecurityDomain = builder.trustedSecurityDomain;
+        this.scheduledExecutorService = builder.scheduledExecutorService;
```

Comment:
If none was set, there should probably be a default single-thread one created, else (by my reading) we have a potential NPE later in the day.